acquia/mc-cs-plugin-custom-objects
# MAUT-11289 : Segments: New (not)between date filter

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | https://acquia.atlassian.net/browse/MAUT-11289 



#### Description:
This PR adds new between and not between operators for the date fields and allows users to set 2 absolute dates like this:

<img width="1098" alt="image" src="https://github.com/acquia/mc-cs-plugin-custom-objects/assets/48244990/98262c07-5d13-4361-9f2c-807859f219c6">

The validation is also in place. This is how it looks like when one of the date ranges is empty:
<img width="1092" alt="image" src="https://github.com/acquia/mc-cs-plugin-custom-objects/assets/48244990/54e6d3d2-141c-4f95-ab0a-3d102aac1817">



#### Steps to test this PR:


1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check that the new not/between operators are there for the date fields but not for other field types where it doesn't make sense.
3. There is the new dual-input for for the filter value only when a not/between operator for the date field is selected.
4. Test that the contacts are added to the segment based on a custom object date field 
5. Test the filter field validation. Each of the 2 new inputs must be a date in the format as the calendar sets it. Example: Jan 5, 2024. Empty values and other values should be rejected.
6. Test both filters, the between but also the not between.
